runtime/builtin_types.go

```diff
@@ -570,6 +571,38 @@ func builtinRepr(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
 	return s.ToObject(), nil
 }
 
+func builtinRound(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	argc := len(args)
+	expectedTypes := []*Type{ObjectType, ObjectType}
+	var raised *BaseException
+	if argc == 1 {
+		expectedTypes = expectedTypes[:1]
+	}
+	if raised = checkFunctionArgs(f, "round", args, expectedTypes...); raised != nil {
+		return nil, raised
+	}
+	ndigits := 0
+	if argc > 1 {
+		ndigits, raised = IndexInt(f, args[1])
```

Comment:
To my surprise, the CPython's `round` function do accept an object with `__index__` method as second parameter. Added a test case for this.